### PR TITLE
docs: Update code level documentation

### PR DESCRIPTION
**Pull request by Rhuagh**
This change is [Reviewable](https://reviewable.io/reviews/omni-viral/xfg-rs/2)

### CODE REVIEW

File: src/pass/build.rs [hunk 1]
```diff
@@ -341,8 +370,8 @@ where
 }
 
 fn push_vertex_desc<B>(attributes: &[pso::Element<Format>], stride: pso::ElemStride, pipeline_desc: &mut pso::GraphicsPipelineDesc<B>)
-where
-    B: Backend,
+    where
+        B: Backend,
```
Comment (omni-viral) [inline src/pass/build.rs]: Is this a new formatting style rustfmt uses?

File: src/graph/build.rs [hunk 2]
```diff
@@ -18,21 +17,27 @@ use attachment::{Attachment, AttachmentImageViews, ColorAttachment, ColorAttachm
...
     FramebufferError,
+    /// Shader module creation error
     ShaderError(ShaderError),
+    /// If no presentation render target is set
     PresentationAttachmentNotSet,
+    /// If no backbuffer is set
     BackbufferNotSet,
+    /// Allocation errors as returned by the `allocator` function given to `GraphBuilder::build`
     AllocationError(E),
+    /// Any other errors encountered during graph builder, such as graphics pipeline creation errors
```
Comment (omni-viral) [inline src/graph/build.rs]: This is more like unknown error from backend. 
Pipeline cration error will cause panic as pass building code is expected to always configure it properly.

**Comment by Rhuagh** [comment]
Review status: 0 of 10 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/graph/build.rs, line 40 at r1](https://reviewable.io:443/reviews/omni-viral/xfg-rs/2#-L3jaM5hUA_PpqcuG912-r1-40:-L3lLYMSMv6jGo-2gSF0:btaypve) ([raw file](https://github.com/omni-viral/xfg-rs/blob/b1aa35f9fb548fa58e29041c2b8d1ffa5cff6fde/src/graph/build.rs#L40)):*
<details><summary><i>Previously, omni-viral (Zakarum) wrote…</i></summary><blockquote>

This is more like unknown error from backend. 
Pipeline cration error will cause panic as pass building code is expected to always configure it properly.
</blockquote></details>

Should we provide  a traceback for it? `Option<Box<Error>>` maybe?

---

*[src/pass/build.rs, line 374 at r1](https://reviewable.io:443/reviews/omni-viral/xfg-rs/2#-L3jaM5hUA_PpqcuG915-r1-374:-L3lLkRv_8WxFXJOYj3N:b6qgev6) ([raw file](https://github.com/omni-viral/xfg-rs/blob/b1aa35f9fb548fa58e29041c2b8d1ffa5cff6fde/src/pass/build.rs#L374)):*
<details><summary><i>Previously, omni-viral (Zakarum) wrote…</i></summary><blockquote>

Is this a new formatting style rustfmt uses?
</blockquote></details>

No ,that's just a messup by me.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/omni-viral/xfg-rs/2)*

**Comment by Rhuagh** [comment]
Review status: 0 of 10 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/graph/build.rs, line 40 at r1](https://reviewable.io:443/reviews/omni-viral/xfg-rs/2#-L3jaM5hUA_PpqcuG912-r1-40:-L3lMaWI8VIvR5EiDpIk:b-896fix) ([raw file](https://github.com/omni-viral/xfg-rs/blob/b1aa35f9fb548fa58e29041c2b8d1ffa5cff6fde/src/graph/build.rs#L40)):*
<details><summary><i>Previously, Rhuagh (Simon Rönnberg) wrote…</i></summary><blockquote>

Should we provide  a traceback for it? `Option<Box<Error>>` maybe?
</blockquote></details>

Done.

---

*[src/pass/build.rs, line 374 at r1](https://reviewable.io:443/reviews/omni-viral/xfg-rs/2#-L3jaM5hUA_PpqcuG915-r1-374:-L3lMZI2J4ItHTuh-G0H:b-896fix) ([raw file](https://github.com/omni-viral/xfg-rs/blob/b1aa35f9fb548fa58e29041c2b8d1ffa5cff6fde/src/pass/build.rs#L374)):*
<details><summary><i>Previously, Rhuagh (Simon Rönnberg) wrote…</i></summary><blockquote>

No ,that's just a messup by me.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/omni-viral/xfg-rs/2)*

**Comment by omni-viral** [comment]
Review status: 0 of 12 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/graph/build.rs, line 40 at r1](https://reviewable.io:443/reviews/omni-viral/xfg-rs/2#-L3jaM5hUA_PpqcuG912-r1-40:-L3lWJ-wl6VNH1XFWMmp:b-1duty7) ([raw file](https://github.com/omni-viral/xfg-rs/blob/b1aa35f9fb548fa58e29041c2b8d1ffa5cff6fde/src/graph/build.rs#L40)):*
<details><summary><i>Previously, Rhuagh (Simon Rönnberg) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Huh. What I said is wrong. You were right :smile: 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/omni-viral/xfg-rs/2)*

**Comment by omni-viral** [comment]
Review status: 0 of 12 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/graph/build.rs, line 40 at r1](https://reviewable.io:443/reviews/omni-viral/xfg-rs/2#-L3jaM5hUA_PpqcuG912-r1-40:-L3lWUXTYvi3D09JP2Uv:b-d57dx9) ([raw file](https://github.com/omni-viral/xfg-rs/blob/b1aa35f9fb548fa58e29041c2b8d1ffa5cff6fde/src/graph/build.rs#L40)):*
<details><summary><i>Previously, omni-viral (Zakarum) wrote…</i></summary><blockquote>

Huh. What I said is wrong. You were right :smile: 
</blockquote></details>

It indeed goes from `CreationError::Other` returned by `Device::create_graphics_pipelines`.
I was confused because of `unwrap` which is actually applied to `Option` that contains `Result`.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/omni-viral/xfg-rs/2)*

**Comment by omni-viral** [comment]
:lgtm:

---

Review status: 0 of 12 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/omni-viral/xfg-rs/2#-:-L3lWx2pzmXW4DSfUwks:bnfp4nl)*

**Pull request merged**